src/renderers/shared/fiber/ReactFiberBeginWork.js

```diff
@@ -296,7 +296,41 @@ module.exports = function<T, P, I, TI, C>(
     if (!coroutine) {
       throw new Error('Should be resolved by now');
     }
-    reconcileChildren(current, workInProgress, coroutine.children);
+
+    const nextChildren = coroutine.children;
+    const priorityLevel = workInProgress.pendingWorkPriority;
+    // At this point any memoization is no longer valid since we'll have changed
+    // the children.
+    workInProgress.memoizedProps = null;
+    if (!current) {
+      workInProgress.stateNode = mountChildFibersInPlace(
+        workInProgress,
+        workInProgress.stateNode,
+        nextChildren,
+        priorityLevel
+      );
+    } else if (current.child === workInProgress.child) {
```

Comment:
Shouldn't this be `current.stateNode === workInProgress.stateNode`?
I changed it, and tests still pass.
Is this unobservable, or are we missing a test?